pkg/controller/storagecluster/storagecluster.go

```diff
@@ -278,18 +277,27 @@ func (c *Controller) waitingForMigrationApproval(cluster *corev1.StorageCluster)
 
 func (c *Controller) validate(cluster *corev1.StorageCluster) error {
 	if err := c.validateK8sVersion(); err != nil {
+		cluster.Status.Phase = string(corev1.ClusterOperationFailed)
 		return err
 	}
 	if err := c.validateSingleCluster(cluster); err != nil {
+		cluster.Status.Phase = string(corev1.ClusterOperationFailed)
 		return err
 	}
 	if err := c.validateCustomAnnotations(cluster); err != nil {
+		cluster.Status.Phase = string(corev1.ClusterOperationFailed)
 		return err
 	}
 	if err := c.validateCloudStorageLabelKey(cluster); err != nil {
+		cluster.Status.Phase = string(corev1.ClusterOperationFailed)
+		return err
+	}
+	if err := c.validatePreflightCheck(cluster); err != nil {
```

Comment:
Let's check this after `Driver.Validate` so we call the expensive cloud APIs at the very end.